### ISSUE TITLE
Fix all fortitude C131 issues

### ISSUE THREAD

**Issue by awvwgk** [tblite/tblite#362]
Missing default accessibility statements for modules.

<details>

````
src/tblite/coulomb.f90:24:1: C131 module 'tblite_coulomb' missing default accessibility statement
   |
23 | !> Proxy module for handling Coulomb-type interactions
24 | module tblite_coulomb
   | ^^^^^^^^^^^^^^^^^^^^^ C131
25 |    use tblite_coulomb_cache, only : coulomb_cache
26 |    use tblite_coulomb_charge, only : coulomb_charge_type, coulomb_kernel, &
27 |       & effective_coulomb, new_effective_coulomb, &
   |

src/tblite/data.f90:24:1: C131 module 'tblite_data' missing default accessibility statement
   |
23 | !> Proxy module for providing access to element data.
24 | module tblite_data
   | ^^^^^^^^^^^^^^^^^^ C131
25 |    use tblite_data_spin, only : get_spin_constant
26 |    implicit none
   |

src/tblite/param/post_processing/type.f90:20:1: C131 module 'tblite_param_post_processing_type' missing default accessibility statement
   |
19 | !> Defines the abstract base post-processing record type
20 | module tblite_param_post_processing_type
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ C131
21 |    use tblite_param_serde, only : serde_record
22 |    implicit none
   |

src/tblite/xtb.f90:27:1: C131 module 'tblite_xtb' missing default accessibility statement
   |
25 | !> #tblite_xtb_ipea1. The main entry point for performing calculations is
26 | !> provided with #tblite_xtb_singlepoint::xtb_singlepoint.
27 | module tblite_xtb
   | ^^^^^^^^^^^^^^^^^ C131
28 |    use tblite_xtb_calculator, only : xtb_calculator, new_xtb_calculator, param_h0spec
29 |    use tblite_xtb_gfn1, only : new_gfn1_calculator, gfn1_h0spec, export_gfn1_param
30 |    use tblite_xtb_gfn2, only : new_gfn2_calculator, gfn2_h0spec, export_gfn2_param
   |
````
